In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [8]:
dataset=pd.read_csv(r"D:\dataset.csv")

In [9]:
dataset.head()

Gender Test preparation  Total Marks
0    male             none           14
1  female             none           28
2  female             none           18
3  female             none           48
4  female             none           21

# carryout discriptive analytics on the dataset

In [10]:
dataset.describe()

Total Marks
count    28.000000
mean     32.321429
std      12.454520
min      12.000000
25%      22.500000
50%      33.000000
75%      43.250000
max      50.000000

# check whether male and female students scored the same marks

In [14]:
import scipy.stats as stats
stats.ttest_ind(dataset['Total Marks'][dataset['Gender'] == 'male'],
                 dataset['Total Marks'][dataset['Gender'] == 'female'])

Ttest_indResult(statistic=-0.6940885268848668, pvalue=0.4937822317960763)

solution

take Null Hypothesis Ho as 'the marks scored by male and female students are not same'

df=n-1 = 28-1 = 27 level of significance is 5% = 0.05

then the table value is 2.052

the calculated value is -0.694

Here, |t score|=0.694 < 2.052 then accept the null hypothesis

Also, p value=0.49>0.05 implies accept Ho

which shows that the marks of female and male students were not same

# Whether test preparation helps the students?

In [17]:
import scipy.stats as stats
stats.ttest_ind(dataset['Total Marks'][dataset['Test preparation'] == 'none'],
                 dataset['Total Marks'][dataset['Test preparation'] == 'completed'])

Ttest_indResult(statistic=-3.6143637527769217, pvalue=0.0012671292457320106)

In [ ]:
solution

take the Null Hypothesis Ho as 'the Test preparation does not helps the students' then the alternative hypothesis H1 is 'the test preparation helps the students'

df=n-1 = 28-1 =27 level of significance =5% = 0.05 the corresponding table value = 2.052

calculated value = -3.614 Here ,|t score|=3.614 > 2.052 , then reject Ho and accept alternative hypothesis

Also,p value = 0.00126 < 0.05 implies reject Ho and accept H1

which shows that the test preparation helps the students

 